### ISSUE TITLE
CreateContainerConfigError status on newly deployed pods

### ISSUE THREAD

**Issue by shivansh** [cloudfoundry-incubator/eirini-release#111]
### Description
The newly deployed pods when installing CF after following the [installation instructions](https://github.com/cloudfoundry-incubator/eirini-release#installation) (step 8) are marked with `CreateContainerConfigError` status.

### Steps to reproduce
[Installation instructions](https://github.com/cloudfoundry-incubator/eirini-release#installation)

### What was expected to happen
A successful deployment.

### What actually happened
All the newly started pods were marked with `CreateContainerConfigError` status.

### Additional information (optional)

The logs for all the pods had a similar error -
```
Warning  Failed     4m11s (x6 over 4m53s)  kubelet, <redacted-address>  Error: secrets "secrets-2.16.4-1" not found
```

Not sure if relevant, but the UAA secret version is -

```
$ SECRET=$(kubectl get pods --namespace uaa -o jsonpath='{.items[?(.metadata.name=="uaa-0")].spec.containers[?(.name=="uaa")].env[?(.name=="INTERNAL_CA_CERT")].valueFrom.secretKeyRef.name}')
$ echo $SECRET
secrets-2.17.1-1
```
- - -

### Update (2018-08-13)
The following error corresponding to the eirini pod is more descriptive -
```
  Warning  FailedMount  2m57s (x8 over 4m1s)  kubelet, <redacted-address>  MountVolume.SetUp failed for volume "cf-secrets" : secrets "secrets-2.16.4-1" not found
  Warning  FailedMount  118s                  kubelet, <redacted-address> Unable to mount volumes for pod "eirini-6fc8768c54-7vhkd_scf(678e105e-bdb2-11e9-9382-baf1f4ff5aef)": timeout expired waiting for volumes to attach or mount for pod "scf"/"eirini-6fc8768c54-7vhkd". list of unmounted volumes=[cf-secrets]. list of unattached volumes=[config-map-volume config-volume cf-secrets opi-token-rmh9w]
  Normal   Pulling      105s                  kubelet, <redacted-address>  pulling image "eirini/opi-init@sha256:a17cfbef0a67cdf5ec6af5d96440627d2f5cf893fab70b36300ab05c6f29d682"
  Normal   Pulled       98s                   kubelet, <redacted-address> Successfully pulled image "eirini/opi-init@sha256:a17cfbef0a67cdf5ec6af5d96440627d2f5cf893fab70b36300ab05c6f29d682"
  Normal   Created      98s                   kubelet, <redacted-address>  Created container
  Normal   Started      98s                   kubelet, <redacted-address>  Started container
  Normal   Started      76s (x2 over 80s)     kubelet, <redacted-address> Started container
  Warning  BackOff      73s                   kubelet, <redacted-address>  Back-off restarting failed container
```

**Comment by akshaymankar** [comment]
@shivansh Can you please share status of `secret-generation` job and related pods from the `scf` namespace?

**Comment by shivansh** [comment]
@akshaymankar The extensive logs for all the pods in the `scf` namespace are available [here](https://gist.github.com/shivansh/f84bfa141d873022cf8c37ae2965223f).

The logs for the `secret-generation` job are -
```
Normal  SuccessfulCreate  23m   job-controller  Created pod: secret-generation-1-cqdr8
```

Please find the logs corresponding to the related pods from the `scf` namespace -
`secret-generation-1-cqdr8` -
```
Normal  Scheduled  6m25s  default-scheduler                                       Successfully assigned scf/secret-generation-1-cqdr8 to <redacted-address>
Normal  Pulling    6m24s  kubelet, <redacted-address>  pulling image "docker.io/eirinicf/scf-secret-generation:6ab7d03a7ab023968868409e31e5590450cddaa2"
Normal  Pulled     3m18s  kubelet, <redacted-address>  Successfully pulled image "docker.io/eirinicf/scf-secret-generation:6ab7d03a7ab023968868409e31e5590450cddaa2"
Normal  Created    3m11s  kubelet, <redacted-address>  Created container
Normal  Started    3m11s  kubelet, <redacted-address>  Started container
```

`secret-smuggler-1-w9r7x` -
```
Normal   Scheduled  7m19s                  default-scheduler                                       Successfully assigned scf/secret-smuggler-1-w9r7x to <redacted-address>
Normal   Pulling    7m18s                  kubelet, <redacted-address>  pulling image "eirini/secret-smuggler@sha256:2a406ee4abc7317432269a77056a87a550773038d68440267735851361b784a7"
Normal   Pulled     4m21s                  kubelet, <redacted-address>  Successfully pulled image "eirini/secret-smuggler@sha256:2a406ee4abc7317432269a77056a87a550773038d68440267735851361b784a7"
Warning  Failed     2m56s (x9 over 4m21s)  kubelet, <redacted-address>  Error: secrets "secrets-2.16.4-2" not found
Normal   Pulled     2m43s (x9 over 4m21s)  kubelet, <redacted-address>  Container image "eirini/secret-smuggler@sha256:2a406ee4abc7317432269a77056a87a550773038d68440267735851361b784a7" already present on machine
Normal   Created    2m43s                  kubelet, <redacted-address>  Created container
Normal   Started    2m42s                  kubelet, <redacted-address>  Started container
```

The logs for `api-group-0` pod are interesting -
```
Normal   Scheduled  3m16s                  default-scheduler                                       Successfully assigned scf/api-group-0 to <redacted-address>
Normal   Pulled     3m15s                  kubelet, <redacted-address>  Container image "docker.io/eirinicf/scf-api-group:8167da745730349f7fa99b4cfc486db72e42aac4" already present on machine
Normal   Created    3m15s                  kubelet, <redacted-address>  Created container
Normal   Started    3m15s                  kubelet, <redacted-address>  Started container
Normal   Pulled     3m15s                  kubelet, <redacted-address>  Container image "docker.io/eirinicf/scf-loggregator-agent:069c32a42dc846117cc319a5c44124018b281a9d" already present on machine
Normal   Created    3m15s                  kubelet, <redacted-address>  Created container
Normal   Started    3m15s                  kubelet, <redacted-address>  Started container
Warning  Unhealthy  3m13s                  kubelet, <redacted-address>  Readiness probe failed: Waiting for monit to start
Warning  Unhealthy  2m53s (x2 over 3m3s)   kubelet, <redacted-address>  Readiness probe failed:
Warning  Unhealthy  2m13s (x4 over 2m43s)  kubelet, <redacted-address>  Readiness probe failed: Waiting for monit to be ready
```

The eventual state of pods in the scf namespace is as below -
```
NAME                             READY   STATUS             RESTARTS   AGE
adapter-0                        2/2     Running            0          12m
api-group-0                      2/2     Running            0          6m34s
bits-65f4f94c88-xtp6g            0/1     CrashLoopBackOff   7          17m
blobstore-0                      2/2     Running            0          11m
cc-clock-0                       2/2     Running            0          17m
cc-uploader-0                    2/2     Running            0          17m
cc-worker-0                      2/2     Running            0          17m
cf-usb-group-0                   1/1     Running            0          8m14s
doppler-0                        2/2     Running            0          11m
eirini-fcffbb95f-pqsfs           1/1     Running            0          17m
locket-0                         2/2     Running            0          17m
log-api-0                        2/2     Running            0          10m
log-cache-scheduler-0            2/2     Running            0          11m
loggregator-fluentd-4fc2p        2/2     Running            0          17m
mysql-0                          1/1     Running            0          10m
nats-0                           2/2     Running            0          12m
nfs-broker-0                     1/1     Running            0          17m
post-deployment-setup-1-gbx6s    1/1     Running            0          17m
rootfs-patcher-v81.0.0-1-c97s9   0/1     Completed          0          17m
router-0                         1/2     Running            0          11m
routing-api-0                    2/2     Running            0          17m
secret-generation-1-cqdr8        0/1     Completed          0          17m
secret-smuggler-1-w9r7x          0/1     Completed          0          17m
syslog-scheduler-0               2/2     Running            0          17m
```

The `bits` pod restarts continuously, and the corresponding logs are -
```
Normal   Scheduled  24m                   default-scheduler                                       Successfully assigned scf/bits-65f4f94c88-xtp6g to <redacted-address>
Normal   Pulling    24m                   kubelet, <redacted-address>  pulling image "flintstonecf/eirinifs-downloader:2.27.0-dev.6"
Normal   Pulled     24m                   kubelet, <redacted-address>  Successfully pulled image "flintstonecf/eirinifs-downloader:2.27.0-dev.6"
Normal   Created    24m                   kubelet, <redacted-address>  Created container
Normal   Started    24m                   kubelet, <redacted-address>  Started container
Normal   Pulled     20m (x4 over 21m)     kubelet, <redacted-address>  Successfully pulled image "flintstonecf/bits-service:2.28.0-dev.7"
Normal   Created    20m (x4 over 21m)     kubelet, <redacted-address>  Created container
Normal   Started    20m (x4 over 21m)     kubelet, <redacted-address>  Started container
Warning  BackOff    9m56s (x52 over 20m)  kubelet, <redacted-address>  Back-off restarting failed container
Normal   Pulling    4m54s (x9 over 23m)   kubelet, <redacted-address>  pulling image "flintstonecf/bits-service:2.28.0-dev.7"
```

**Comment by JulzDiverse** [comment]
@shivansh The `CreateContainerConfigError` status is normal as most of the pods rely on a job that needs to finish first (`secret-generation` job). So eventually all pods need to come up (this takes usually about 15 mins). 
The question is why the bits-service isn't running for you. Could you share/provide the bits-service logs too? Also, what is the size of your cluster?

**Comment by shivansh** [comment]
@JulzDiverse The logs for `bits-service` are -
```
Name:                     bits
Namespace:                scf
Labels:                   <none>
Annotations:              <none>
Selector:                 name=bits
Type:                     LoadBalancer
IP:                       100.69.95.96
LoadBalancer Ingress:     a086b5089c4a111e988d60a59d653323-806689300.eu-central-1.elb.amazonaws.com
Port:                     bits  6666/TCP
TargetPort:               6666/TCP
NodePort:                 bits  31311/TCP
Endpoints:
Session Affinity:         None
External Traffic Policy:  Cluster
Events:
  Type    Reason                Age   From                Message
  ----    ------                ----  ----                -------
  Normal  EnsuringLoadBalancer  23m   service-controller  Ensuring load balancer
  Normal  EnsuredLoadBalancer   23m   service-controller  Ensured load balancer
```

The detailed logs for the `bits` pod might be suggesting where the problem lies -

```
2019-08-22T06:03:34.266Z	INFO	bitsgo/main.go:39	Logging level	{"log-level": "debug"}
{"level":"info","ts":1566453814.2667656,"caller":"bitsgo/main.go:44","msg":"Config file uses deprecated \"secret\" property. Please consider using \"signing_keys\" instead."}
{"level":"info","ts":1566453814.2668743,"caller":"bitsgo/factory.go:303","msg":"Creating Webdav blobstore","public-endpoint":"https://blobstore.example.mydomain.com","private-endpoint":"https://blobstore-blobstore.scf.svc.cluster.local:4443"}
{"level":"info","ts":1566453814.2669423,"caller":"webdav/http_client.go:13","msg":"Creating Http Client","insecure-skip-verify":true,"ca-cert-path":""}
panic: Could not append pemCerts. pemCerts content:

\```

\```

goroutine 1 [running]:
github.com/cloudfoundry-incubator/bits-service/blobstores/webdav.NewHttpClient(0x0, 0x0, 0xc000378301, 0x36)
	/tmp/build/37652675/bits-service-release/src/github.com/cloudfoundry-incubator/bits-service/blobstores/webdav/http_client.go:18 +0x388
github.com/cloudfoundry-incubator/bits-service/blobstores/webdav.NewBlobstore(0xc000376560, 0xc, 0xc000378300, 0x36, 0xc00037c1e0, 0x26, 0xc00037c210, 0x2a, 0x1, 0xc0003763b0, ...)
	/tmp/build/37652675/bits-service-release/src/github.com/cloudfoundry-incubator/bits-service/blobstores/webdav/webdav_blobstore.go:30 +0x84
main.createAppStashBlobstore(0xc000376518, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0003a7ab0, 0x0, 0x0, ...)
	/tmp/build/37652675/bits-service-release/src/github.com/cloudfoundry-incubator/bits-service/cmd/bitsgo/factory.go:309 +0x1063
main.main()
	/tmp/build/37652675/bits-service-release/src/github.com/cloudfoundry-incubator/bits-service/cmd/bitsgo/main.go:49 +0x46f
```

I have a few speculations as to where things are going wrong -
* The public endpoint `https://blobstore.example.mydomain.com` is not accessible
* `pemCerts` is empty

I was careful in closely following the instructions mentioned in the repo, is there any other step that I need to follow to resolve the issue (apart from making the aforementioned public endpoint accessible) ?

**Comment by mnitchev** [comment]
Hi @shivansh,

The public blobstore url for bits-service [is set here](https://github.com/cloudfoundry-incubator/bits-service-release/blob/master/helm/bits/templates/bits.yaml#L55-L59), and I assume you're using `loadbalanced: true` based on the kube logs you provided. It looks like somehow your [`bits.DOMAIN` here](https://github.com/cloudfoundry-incubator/eirini-release/blob/master/values.yaml#L43) is different from the rest of your CF (example.mydomain.com), and you also haven't set the `BITS_TLS_KEY` and `BITS_TLS_CRT` when deploying, which is why `pemCerts` is empty.
Can you verify that this is the case?

**Comment by julz** [comment]
Hi @shivansh - Im going to close this issue since it has been inactive for a while. Please feel free to re-open or create a new issue if you're still having trouble.
